### PR TITLE
LazyRegressionDataset, compute_norms.py, and fix to loaders

### PR DESCRIPTION
**Pull request by gabegrand**
This PR has several major improvements:

- Adds a `LazyRegressionDataset` and `model_type=regression_lazy`, which is like the existing `regression` / MTR pretraining mode, except that it takes a .smi file and computes RDKit descriptors on-the-fly as part of the dataset's `preprocess()` method.
- Adds a script `compute_norms.py` that takes in a .smi file and returns a .json with the mean and std of the descriptors for use in MTR pretraining.
- Fixes a nasty bug affecting `RawTextDataset` where the model inputs included additional non-SMILES characters. See below.

### CODE REVIEW

File: chemberta/utils/raw_text_dataset.py [hunk 1]
```diff
@@ -41,9 +44,9 @@ def __init__(self, tokenizer, file_path: str, block_size: int):
     def __len__(self):
         return self.len
 
-    def preprocess(self, text):
+    def preprocess(self, feature_dict):
         batch_encoding = self.tokenizer(
-            str(text),
+            feature_dict["text"],
```
Comment (gabegrand) [inline chemberta/utils/raw_text_dataset.py:49]: I discovered that the arg to `preprocess` is actually a feature dictionary. Previously, the dataset code was coercing this to a string representation that contained extraneous characters; e.g.:

```
>> print(str(text))
"{'text': 'CCC(CC)(Pc1ccc(C)cc1CO)c1cc(OC)ccc1OCOC'}"
```

File: chemberta/data/pubchem_descriptors_sample_1k_normalization_values_207.json [hunk 2]
```diff
@@ -0,0 +1 @@
...62276075862127, 0.10914124661012291, 0.0891284456038114, 0.9416988407216558, 0.7968785326360313, 1.2210078745032462, 0.14753138395947027, 0.8459317261873186, 0.7140260194697382, 0.054717318793659306, 0.031622760758621284, 0.044698926049582074, 1.1304949179453359, 0.0, 1.177711604018516, 0.03162276075862126, 0.0, 0.044698926049581915, 0.4002777589954339, 0.8856975419473017, 0.20376210755180174, 0.12933459764834604, 1.2018024854771152, 0.12933459764834596, 0.12553686011284088, 0.2523952118852928, 0.2001951951326321, 0.0, 0.0, 0.30099352752196873, 0.2552375899204281, 0.054717318793659896, 0.054717318793659514, 0.52315687181949, 0.17066993679739162, 0.2074029627607446, 0.21298516379624982, 0.19953342417032294, 0.16304951303901577, 0.054717318793659785, 0.08912844560381099, 0.044698926049582095, 0.5130264408994892, 0.18710955336362645, 0.18710955336362645, 0.10943463758731921, 0.07068229607171143, 0.3691129264338621, 0.271502031286384, 0.1434543052552089, 0.0, 0.4744132278442473, 0.5481904093808613, 0.30981910814483865, 0.2969718781307522, 0.15952702685037234, 0.13746918438520642, 0.0705687957237828, 0.24861527970953778, 0.0, 0.2654559535208344, 1.6328078444361518, 0.22055666355482306]}
```
Comment (walid0925) [inline chemberta/data/pubchem_descriptors_sample_1k_normalization_values_207.json:1]: how do we know which one of these files to use?

File: chemberta/utils/raw_text_dataset.py (hunk 1 above)
Comment (walid0925) [reply chemberta/utils/raw_text_dataset.py:49]: very good catch

**Review by walid0925 — APPROVED** [review]
lgtm! great work on finding the dictionary bug

File: chemberta/data/pubchem_descriptors_sample_1k_normalization_values_207.json (hunk 2 above)
Comment (gabegrand) [reply chemberta/data/pubchem_descriptors_sample_1k_normalization_values_207.json:1]: yeah so I discovered that the newer `rdkit 2021.03.1` release actually added some additional descriptors; before, there were 200 and now there are 208 (in both cases, we remove `Ipc` which I found takes a while to compute).

**Pull request merged**